lib/src/layout/window.dart

```diff
@@ -95,7 +96,20 @@ class _MacosWindowState extends State<MacosWindow> {
     late Color sidebarBackgroundColor;
     Color dividerColor = theme.dividerColor;
 
-    if (!theme.brightness.isDark) {
+    final isMac = !kIsWeb && defaultTargetPlatform == TargetPlatform.macOS;
+
+    // Respect the sidebar color override from parent if one is given
+    if (widget.sidebar?.decoration?.color != null) {
+      sidebarBackgroundColor = widget.sidebar!.decoration!.color!;
+    } else if (isMac &&
+        MediaQuery.of(context).platformBrightness.isDark ==
+            theme.brightness.isDark) {
+      // Only show blurry, transparent sidebar when platform brightness and app
+      // brightness are the same, otherwise it looks awful. Also only make the
+      // sidebar transparent on native Mac, or it will just be flat black or
+      // white.
+      sidebarBackgroundColor = Colors.transparent;
+    } else if (!theme.brightness.isDark) {
```

Comment:
Sorry to but in again, I just realized the suggestion I sent was an older version of the code. Since we're now checking for `widget.sidebar?.decoration?.color` in the first `if` branch, we don't need to consider it when falling back to the theme. The last two branches can be collapsed into single else clause with a ternary statement:

```suggestion
    } else {
        sidebarBackgroundColor = CupertinoColors.systemGrey6.color;
    }
```

Or if you prefer without the ternary:

```suggestion
    } else if (!theme.brightness.isDark) {
        sidebarBackgroundColor = CupertinoColors.systemGrey6.color;
    } else {
        sidebarBackgroundColor = CupertinoColors.tertiarySystemBackground.darkColor;
    }
```

Both changes are supposed to replace lines 112 - 118 and haven't been run through `dartfmt` because github's suggestion system is very annoying to use.

Reply:
@mj-shifu what are your thoughts on @jmatth 's feedback here?

Reply:
I agree with @jmatth 's feedback. But because GitHub wouldn't let me merge the changes directly, I created a new commit to fix it.